subprojects/ear/src/test/groovy/org/gradle/plugins/ear/EarPluginTest.groovy

```diff
@@ -103,7 +103,7 @@ class EarPluginTest extends AbstractProjectBuilderSpec {
 
         then:
         task instanceof Ear
-        task dependsOn(JavaPlugin.CLASSES_TASK_NAME)
+        task dependsOn(JavaPlugin.CLASSES_TASK_NAME, JavaPlugin.COMPILE_JAVA_TASK_NAME)
```

Comment:
What is the exact meaning of having to change these assertions?
In practice, the logical dependency chain should be ear -> classes -> compileJava
With the change, it seems like we flatten some transitive dependencies one level up.